spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,96 @@ def set_tabs_test(option)
       end
     end
   end
+
+  describe "#cmd_sessions" do
+    before(:each) do
+      allow(driver).to receive(:active_session=)
+      
+    end
+
+    context "with no sessions" do
+      it "should show an empty table" do
+        core.cmd_sessions
+        expect(@output).to eq([
+          'Active sessions',
+          '===============',
+          '',
+          'No active sessions.'
+        ])
+      end
+    end
+
+    context "filtering with search results" do
+      before do
+        allow(framework).to receive(:sessions).and_return(sessions)
+        allow(double "Session").to receive(:kill)
+      end
+
+      let(:sessions) do 
+        {
+          1 => double('Session', last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "sesh1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel"),
```

Comment:
Not tested: I believe you should be able to use `instance_double(::Msf::Session, ...)` here. `instance_double` is nicer as it will verify your mocked calls against a real class instance - so if you mock values that are deleted from the real object your tests will start failing. Whilst a mock would continue to work.

This should work the methods that are called aren't defined with metaprogramming/method_missing logic


> In RSpec 3.0 and higher, instance_double is an implementation of verifying doubles. Verifying doubles are defined in RSpec as "a stricter alternative to normal doubles that provide guarantees about what is being verified. When using verifying doubles, RSpec will check that the methods being stubbed are actually present on the underlying object, if it is available."
> https://www.simplybusiness.co.uk/about-us/tech/2020/05/rspec-instance-doubles/